src/main/scala/org/clulab/wm/eidos/groundings/grounders/EidosOntologyGrounder.scala

```diff
@@ -55,37 +66,91 @@ abstract class EidosOntologyGrounder(val name: String, val domainOntology: Domai
     nodes.filter(node => nodePatternsMatch(s, node.patterns)).map(node => SingleOntologyNodeGrounding(node.namer, 1.0f))
   }
 
-  def nodePatternsMatch(s: String, patterns: Option[Array[Regex]]): Boolean = {
-    patterns match {
+  def nodePatternsMatch(string: String, patternsOpt: Option[Array[Regex]]): Boolean = {
+    patternsOpt match {
       case None => false
-      case Some(rxs) =>
-        for (r <- rxs) {
-          if (r.findFirstIn(s).nonEmpty) return true
-        }
-        false
+      case Some(patterns) => patterns.exists(_.findFirstIn(string).nonEmpty)
     }
   }
 
-  def groundPatternsThenEmbeddings(text: String, patterns: Seq[ConceptPatterns], embeddings: Seq[ConceptEmbedding]): MultipleOntologyGrounding = {
-    groundPatternsThenEmbeddings(text, text.split(" +"), patterns, embeddings)
+  // For Regex Matching
+  def nodesExampleMatched(string: String, nodes: Seq[ConceptExamples]): Map[Namer, Float] = {
+    (
+      for (node <- nodes)
+      yield node.namer -> nodeExamplesMatch(string, node.examples)
+    ).toMap
   }
-  def groundPatternsThenEmbeddings(splitText: Array[String], patterns: Seq[ConceptPatterns], embeddings: Seq[ConceptEmbedding]): MultipleOntologyGrounding = {
-    groundPatternsThenEmbeddings(splitText.mkString(" "), splitText, patterns, embeddings)
+
+  def nodeExamplesMatch(string: String, examples: Option[Array[String]]): Float = {
+    examples match {
+      case None => string.length.toFloat
+      case Some(examples) =>
+        val lowerString = string.toLowerCase // just once for all examples
+        examples
+            .map { example => new EditDistance().score(lowerString, example.toLowerCase) }
+            .min
+            .toFloat
+    }
+  }
+
+  def groundExamplesThenEmbeddings(text: String, examples: Seq[ConceptExamples], embeddings: Seq[ConceptEmbedding]): MultipleOntologyGrounding = {
+    ???
```

Comment:
I am to blame for this line of code.  I'm guessing that it was added as a placeholder, but that it wasn't used in the end.  The method doesn't seem to ever be called, but it should probably be removed, because the ??? will throw an exception and it's an accident waiting to happen.  I may do it depending on what else there is.

Reply:
I think it's fine to remove this. It was a placeholder when you added support for examples, but the way the algorithm developed just added those into an existing function instead of using this new one.